content/browser/web_contents/web_contents_impl.cc

```diff
@@ -1097,9 +1097,13 @@ std::unique_ptr<WebContentsImpl> WebContentsImpl::CreateWithOpener(
   std::unique_ptr<WebContentsImpl> new_contents(
       new WebContentsImpl(params.browser_context));
 
-  // Forward flag indicating that this web-contents is being created
-  // for a replay.io recording.
-  if (params.record_replay_for_recording) {
+  // Forward params flag (and RECORD_ALL_CONTENT env var) indicating that this
+  // web-contents is being created for a replay.io recording.  Ideally we'd
+  // modify the params here to avoid directly setting on `new_contents`, but
+  // it's const here.
+  std::unique_ptr<base::Environment> env(base::Environment::Create());
+  bool has_record_all_content = env->HasVar("RECORD_ALL_CONTENT");
```

Comment:
move the env var check here (to the browser process.)  I _think_ I'd rather include it in the constructor for `WebContents::CreateParams` instead.  That would also make it so we don't need to directly set `new_contents->record_replay_for_recording_`, and the `WebContentsImpl::Init` method can access it from the params instead.